SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/CodeGenerator/Unreal/InterfaceGenerator.cs

```diff
@@ -0,0 +1,197 @@
+using Improbable.Codegen.Base;
+using Improbable.CodeGen.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Improbable.CodeGen.Unreal
+{
+    public static class InterfaceGenerator
+    {
+        public static string ClassName = $"ExternalSchemaInterface";
+        public static string HeaderPath = $"ExternalSchemaInterface.h";
+        public static string SourceFile = $"ExternalSchemaInterface.cpp";
+
+        public static List<GeneratedFile> GenerateInterface(List<TypeDescription> componentTypes, Bundle bundle)
+        {
+            return new List<GeneratedFile>
+            {
+                new GeneratedFile(HeaderPath, GenerateInterfaceHeader(componentTypes, bundle)),
+                new GeneratedFile(SourceFile, GenerateInterfaceSource(componentTypes, bundle))
+            };
+        }
+
+        public static string GenerateInterfaceHeader(List<TypeDescription> componentTypes, Bundle bundle)
+        {
+            var builder = new StringBuilder();
+
+            builder.AppendLine($@"#pragma once
+
+#include ""CoreMinimal.h""
+#include ""Connection/SpatialWorkerConnection.h""
```

Comment:
Please remove this and forward declare the relevant classes, you may need to move the constructor definition to the cpp for that.